File: ios/ViewController.m
//
//  ViewController.m
//  BasicPlayer-ObjC
//
//  Created by Michael Salvador on 8/3/21.
//

#import "ViewController.h"

#define videoUrlString @"https://live.impresa.pt/live/sic/sic.m3u8"
#define posterUrlString @"https://d3el35u4qe4frz.cloudfront.net/bkaovAYt-480.jpg"

@interface ViewControllerImpresa ()

@end

@implementation ViewControllerImpresa

- (void)viewDidLoad {
    [super viewDidLoad];

    // Set the view's background color to black for better contrast.
    self.view.backgroundColor = UIColor.blackColor;

    // Set up the player.
    [self setUpPlayer];
}

-(void)setFile:(NSString*)file {
    NSURL *url = [NSURL URLWithString:file];
    urlFile = url;
}

/**
 Sets up the player with a simple configuration.
 */
- (void)setUpPlayer {
    NSLog(@"Aqui 2");
    NSURL *videoUrl = [NSURL URLWithString:videoUrlString];
    NSURL *posterUrl = [NSURL URLWithString:posterUrlString];

    // First, use the JWPlayerItemBuilder to create a JWPlayerItem that will be used by the player configuration.
    JWError *playerItemError;
    playerItemBuilder = [[JWPlayerItemBuilder alloc] init];
    [playerItemBuilder file:videoUrl];
    [playerItemBuilder posterImage:posterUrl];
    JWPlayerItem *playerItem = [playerItemBuilder buildAndReturnError:&playerItemError];
    if (playerItemError != nil) {
        // Handle error
        return;
    }

    // Second, create a player config with the created JWPlayerItem.
    JWError *configError;
    JWPlayerConfigurationBuilder *configBuilder = [[JWPlayerConfigurationBuilder alloc] init];
    [configBuilder playlist:@[playerItem]];
    JWPlayerConfiguration *config = [configBuilder buildAndReturnError:&configError];
    if (configError != nil) {
        // Handle error
        return;
    }

    // Lastly, use the created JWPlayerConfiguration to set up the player.
    [self.player configurePlayerWith:config];
}


@end

File: ios/ViewController.swift
import UIKit
import JWPlayerKit

/**
 Provides minimal setup for a JWPlayerViewController.
 */
class ViewController: JWPlayerViewController {

    private let videoUrlString = "https://live.impresa.pt/live/sic/sic.m3u8"
    private let posterUrlString = "http://d3el35u4qe4frz.cloudfront.net/bkaovAYt-480.jpg"

    override func viewDidLoad() {
        super.viewDidLoad()

        // Set the view's background color to black for better contrast.
        view.backgroundColor = .black

        // Set up the player.
        setUpPlayer()
        print("MyViewController loaded...")

    }

    /**
     Sets up the player with a simple configuration.
     */
    private func setUpPlayer() {
        let videoUrl = URL(string:videoUrlString)!
        let posterUrl = URL(string:posterUrlString)!

        // First, use the JWPlayerItemBuilder to create a JWPlayerItem that will be used by the player configuration.
        let playerItembuilder = JWPlayerItemBuilder()
            .file(videoUrl)
            .posterImage(posterUrl)
        var playerItem: JWPlayerItem!
        do {
            playerItem = try playerItembuilder.build()
        } catch {
            // Handle player item build failure
            print(error.localizedDescription)
            return
        }

        // Second, create a player config with the created JWPlayerItem.
        let configBuilder = JWPlayerConfigurationBuilder()
            .playlist([playerItem])
            .autostart(true)
        var config: JWPlayerConfiguration!
        do {
            config = try configBuilder.build()
        } catch {
            // Handle player item build failure
            print(error.localizedDescription)
            return
        }

        // Lastly, use the created JWPlayerConfiguration to set up the player.
        player.configurePlayer(with: config)
    }
}


